src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3735,6 +3772,48 @@ class CgroupUtils(object):
                 raise
         return self.assigned_resources is not None
 
+    def add_jobid_to_cgroup_jobs(self, jobid):
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'Adding jobid %s to cgroup_jobs' % jobid)
+        with open(self.cgroup_jobs_file, 'a+') as f:
+            f.write(jobid + '\n')
+        return True
+
+    def remove_jobid_from_cgroup_jobs(self, jobid):
+        if os.path.isfile(self.cgroup_jobs_file):
+            with open(self.cgroup_jobs_file, 'r+') as f:
+                jobid_list = f.readlines()
+                f.seek(0)
+                for j in jobid_list:
+                    if jobid != j.strip():
```

Comment:
Or you could read/write the list as a single line...
```
$ python
Python 2.7.5 (default, Aug  4 2017, 00:39:18)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> desc = open('myfile', 'r+')
>>> mylist = desc.readline().split()
>>> type(mylist)
<type 'list'>
>>> print mylist
['123.foo', '456.bar', '789.baz']
>>> newlist = [x for x in mylist if x != '456.bar']
>>> print newlist
['123.foo', '789.baz']
>>> desc.seek(0)
>>> desc.write(' '.join(newlist))
>>> desc.truncate()
>>> desc.close()
>>> exit()
```

Reply:
Since you're using sets, which guarantee unique entries, why not use set.add() and set.discard()?

Reply:
Great idea. Thanks. Done